debugger/src/rtt.rs

```diff
@@ -0,0 +1,388 @@
+use crate::*;
+use anyhow::{anyhow, Result};
+use chrono::Local;
+use num_traits::Zero;
+use probe_rs::Core;
+use probe_rs_rtt::{DownChannel, UpChannel};
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::{
+    fmt,
+    fmt::Write,
+    fs,
+    io::{Read, Seek},
+    str::FromStr,
+};
+use structopt::StructOpt;
+
+/// Used by serde to provide defaults for `RttConfig`
+fn default_channel_formats() -> Vec<RttChannelConfig> {
+    vec![]
+}
+fn default_rtt_timeout() -> usize {
+    500
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
+pub enum DataFormat {
+    String,
+    BinaryLE,
+    Defmt,
+}
+impl FromStr for DataFormat {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let src = s.to_ascii_lowercase();
+        match &src.to_ascii_lowercase()[..] {
+            // A forgiving/case-insensitive match
+            "string" => Ok(Self::String),
+            "binaryle" => Ok(Self::BinaryLE),
+            "defmt" => Ok(Self::Defmt),
+            _ => Err(format!("{} is not a valid format", src)),
+        }
+    }
+}
+impl Default for DataFormat {
+    fn default() -> Self {
+        DataFormat::String
+    }
+}
+
+/// The initial configuration for RTT (Segger Real Time Transfer). This configuration is complimented with the additional information specified for each of the channels in `RttChannel`
```

Comment:
```suggestion
/// The initial configuration for RTT (Real Time Transfer). This configuration is complimented with the additional information specified for each of the channels in `RttChannel`.
```

We should not mention segger as even tho they coined the term, it confuses people into believing they need segger for RTT to work and it really doesn't have anything to do with segger :)